endpoint.bal

```diff
@@ -0,0 +1,536 @@
+            self.accessToken, self.accessTokenSecret, oauthString);
+        if (requestHeaders is error) {
+            error err = error(TWITTER_ERROR,
+                              message = "Error occurred while encoding");
+            return err;
+        } else {
+            var httpResponse = check self.twitterClient->post(resourcePath, request);
+            var response = check handleResponse(httpResponse);
+            return convertToStatus(response);
+        }
+    }
+
+    # Get a User object.
+    #
+    # + userId - The numerical ID of a specific user
+    # + return - If success, returns User object, else returns error
+    @display {label: "Get a user's detail"} 
+    remote function getUser(@display {label: "User Id"} int userId) 
+                            returns @tainted @display {label: "User"} User|error {
+        http:Request request = new;
+        string resourcePath = GET_USER_ENDPOINT;
+        string encodedValue = check url:encode((userId.toString()), UTF_8);
+        string urlParams = USER_ID + encodedValue + AMBERSAND;
+        string oauthString = getOAuthParameters(self.clientId, self.accessToken) + urlParams;
+
+        var requestHeaders = createRequestHeaderMap(request, GET, resourcePath, self.clientId, self.clientSecret,
+            self.accessToken, self.accessTokenSecret, oauthString);
+        if (requestHeaders is error) {
+            error err = error(TWITTER_ERROR,
+                              message = "Error occurred while encoding");
+            return err;
+        } else {
+            map<string> headerMap = requestHeaders;
+            resourcePath = resourcePath + QUESTION_MARK + urlParams;
+            var httpResponse = check self.twitterClient->get(resourcePath, headerMap);
+            var response = check handleResponse(httpResponse);
+            json[] array = <json[]> response;
+            json user = array[0];
+            return convertToUser(user);
+        }
+    }
+
+    # Get a user's followers.
+    #
+    # + userId - The numerical ID of a specific user
+    # + return - If success, returns User object array, else returns error
+    @display {label: "Get a user's followers"} 
+    remote function getFollowers(@display {label: "User Id"} int userId) 
+                                 returns @tainted @display {label: "Array of User"} User[]|error {
+        http:Request request = new;
+        string resourcePath = FOLLOWERS_ENDPOINT;
+        string encodedValue = check url:encode((userId.toString()), UTF_8);
+        string urlParams = USER_ID + encodedValue + AMBERSAND;
+        string oauthString = getOAuthParameters(self.clientId, self.accessToken) + urlParams;
+
+        var requestHeaders = createRequestHeaderMap(request, GET, resourcePath, self.clientId, self.clientSecret,
+            self.accessToken, self.accessTokenSecret, oauthString);
+        if (requestHeaders is error) {
+            error err = error(TWITTER_ERROR,
```

Comment:
I think we need to propagate the original error too. Otherwise we will loose information what actually happened - WDYT?

Reply:
That is better. I will show exact error twitter returns